### PR TITLE
fix: [CDS-45683]: nexus2 connector addition step from nexus2 artifact creation

### PR DESCRIPTION
**Pull request by satyam-harness**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by satyam-harness** [comment]
retrigger titlecheck

**Comment by satyam-harness** [comment]
retrigger titlecheck

**Review by Eric761 — DISMISSED** [review]

File: src/modules/70-pipeline/components/ArtifactsSelection/ArtifactWizard/ArtifactWizard.tsx [hunk 1]
```diff
@@ -114,6 +114,8 @@ function ArtifactWizard({
         return <StepAWSAuthentication name={getString('credentials')} {...newConnectorProps.auth} />
       case ENABLED_ARTIFACT_TYPES.Nexus3Registry:
         return <StepNexusAuthentication name={getString('details')} {...newConnectorProps.auth} />
+      case ENABLED_ARTIFACT_TYPES.Nexus2Registry:
+        return <StepNexusAuthentication name={getString('details')} {...newConnectorProps.auth} />
```
Comment (rameshx) [inline src/modules/70-pipeline/components/ArtifactsSelection/ArtifactWizard/ArtifactWizard.tsx:118]:
```suggestion
      case ENABLED_ARTIFACT_TYPES.Nexus3Registry:
      case ENABLED_ARTIFACT_TYPES.Nexus2Registry:
        return <StepNexusAuthentication name={getString('details')} {...newConnectorProps.auth} />
```

File: src/modules/70-pipeline/components/ArtifactsSelection/ArtifactWizard/ArtifactWizard.tsx [hunk 2]
```diff
@@ -137,6 +139,8 @@ function ArtifactWizard({
         return buildAWSPayload
       case ENABLED_ARTIFACT_TYPES.Nexus3Registry:
         return buildNexusPayload
+      case ENABLED_ARTIFACT_TYPES.Nexus2Registry:
+        return buildNexusPayload
```
Comment (rameshx) [inline src/modules/70-pipeline/components/ArtifactsSelection/ArtifactWizard/ArtifactWizard.tsx:143]:
```suggestion
      case ENABLED_ARTIFACT_TYPES.Nexus3Registry:
      case ENABLED_ARTIFACT_TYPES.Nexus2Registry:
        return buildNexusPayload
```

File: src/modules/70-pipeline/components/ArtifactsSelection/ArtifactWizard/ArtifactWizard.tsx (hunk 1 above)
Comment (satyam-harness) [reply src/modules/70-pipeline/components/ArtifactsSelection/ArtifactWizard/ArtifactWizard.tsx]: addressed

File: src/modules/70-pipeline/components/ArtifactsSelection/ArtifactWizard/ArtifactWizard.tsx (hunk 2 above)
Comment (satyam-harness) [reply src/modules/70-pipeline/components/ArtifactsSelection/ArtifactWizard/ArtifactWizard.tsx]: addressed

**Review by YounixIO — APPROVED** [review]

**Review by pankajsingh-harness — APPROVED** [review]

**Comment by satyam-harness** [comment]
trigger cypress

**Pull request merged**